cloudconfig.go

```diff
@@ -8,24 +8,36 @@ import (
 
 	"github.com/giantswarm/clustertpr"
 	"github.com/giantswarm/clustertpr/node"
+	"k8s.io/client-go/kubernetes"
 )
 
+// Params are the parameters which can be defined by the concrete operator which is using that library.
 type Params struct {
 	Cluster   clustertpr.Cluster
 	Extension Extension
 	Node      node.Node
 }
 
+// internalParams contain both parameters defined by the operator and those defined and used only in
+// this library.
+type internalParams struct {
+	Token string
+	Params
+}
+
 type CloudConfig struct {
 	config   string
-	params   Params
+	params   internalParams
```

Comment:
Why not just keeping the params as they are and adding a private token field? Also, what kind of token is that? I would like the structure and naming to reflect its identity or purpose. Now I know that we talk about the kube admin token, but in half a year some new employee will not.